product_kit_management/wizard/product_kit_wizard.py

```diff
@@ -0,0 +1,99 @@
+from odoo import api, fields, models, Command
+
+
+class ProductKit(models.TransientModel):
+    _name = 'product.kit'
+    _description = "Kit Wizard"
+
+    sale_order_line_id = fields.Many2one(
+        'sale.order.line', string="Sale Order Line",
+        help="Reference to the sale order line"
+    )
+    kit_product_id = fields.Many2one(
+        'product.product', string="Kit Product",
+        help="Main product of kit"
+    )
+    kit_subproducts_ids = fields.One2many(
+        'product.kit.line', 'product_kit_id',
+        string="Sub Products", help="List of sub-products in the kit"
+    )
+
+    def action_confirm(self):
+        """
+        Confirms the changes made in the wizard and updates the sale order line accordingly.
+        - Updates existing kit components with the new quantities and prices.
+        - If a sub-product does not exist in the order, it creates a new sale order line for it.
+        - Adjusts the sequence of sub-products in the sale order.
+        - Updates the total price of the main kit product based on its components.
+        """
+        sale_order_line = self.env['sale.order.line'].browse(self._context.get('active_id'))
+
+        if sale_order_line.is_kit:
+            order = sale_order_line.order_id
+            main_product_sequence = sale_order_line.sequence
+            total_subproduct_price = 0.0
+
+            for index, line in enumerate(self.kit_subproducts_ids, start=1):
+                component = sale_order_line.kit_component_ids.filtered(
+                    lambda c: c.kit_component_id.id == line.product_id.id
+                )
+                if component:
+                    component.write({
+                        'kit_qty': line.quantity,
+                        'price': line.price
+                    })
+                existing_line = order.order_line.filtered(
+                    lambda ol: ol.product_id == line.product_id and 
+                    ol.is_kit_component
+                )
+                if existing_line:
+                    existing_line.write({
+                        'product_uom_qty': line.quantity,
+                        'price_unit': 0.0,  
+                        'sequence': main_product_sequence + index  
+                    })
+                else:
+                    self.env['sale.order.line'].create({
+                        'order_id': order.id,
+                        'product_id': line.product_id.id,
+                        'product_uom_qty': line.quantity,
+                        'price_unit': 0.0,  
```

Comment:
```suggestion
                        'price_unit': 0.0,
```
well just to notify you for extra space...

```suggestion
```